src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/regelverkFørFebruar2025/VilkårResultaterTilAndelTilkjentYtelseMapper.kt

```diff
@@ -0,0 +1,66 @@
+package no.nav.familie.ks.sak.kjerne.beregning.regelverkFørFebruar2025
+
+import no.nav.familie.ks.sak.common.exception.Feil
+import no.nav.familie.ks.sak.common.tidslinje.Periode
+import no.nav.familie.ks.sak.common.util.toYearMonth
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.UtdypendeVilkårsvurdering
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.Vilkår
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.VilkårResultat
+import no.nav.familie.ks.sak.kjerne.beregning.domene.AndelTilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.beregning.domene.SatsPeriode
+import no.nav.familie.ks.sak.kjerne.beregning.domene.TilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.beregning.domene.YtelseType
+import no.nav.familie.ks.sak.kjerne.beregning.domene.hentGyldigSatsFor
+import no.nav.familie.ks.sak.kjerne.beregning.domene.prosent
+import no.nav.familie.ks.sak.kjerne.personident.Aktør
+import java.math.RoundingMode
+
+fun Periode<List<VilkårResultat>>.tilAndelTilkjentYtelse(
+    tilkjentYtelse: TilkjentYtelse,
+    barnAktør: Aktør,
+): AndelTilkjentYtelse {
+    val erDeltBosted =
+        this.verdi.any {
+            it.vilkårType == Vilkår.BOR_MED_SØKER &&
+                it.utdypendeVilkårsvurderinger.contains(UtdypendeVilkårsvurdering.DELT_BOSTED)
+        }
+
+    val antallTimer = this.verdi.single { it.vilkårType == Vilkår.BARNEHAGEPLASS }.antallTimer
+
+    val satsperiode =
+        hentGyldigSatsFor(
+            antallTimer = antallTimer?.setScale(2, RoundingMode.HALF_UP),
+            erDeltBosted = erDeltBosted,
+            stønadFom = fom!!.toYearMonth(),
+            stønadTom = tom!!.toYearMonth(),
+        )
+
+    validerBeregnetPeriode(beløpsperiode = satsperiode, behandlingId = tilkjentYtelse.behandling.id)
+
+    val kalkulertUtbetalingsbeløp = satsperiode.sats.prosent(satsperiode.prosent)
+
+    return AndelTilkjentYtelse(
+        behandlingId = tilkjentYtelse.behandling.id,
+        tilkjentYtelse = tilkjentYtelse,
+        aktør = barnAktør,
+        stønadFom = satsperiode.fom,
+        stønadTom = satsperiode.tom,
+        kalkulertUtbetalingsbeløp = kalkulertUtbetalingsbeløp,
+        nasjonaltPeriodebeløp = kalkulertUtbetalingsbeløp,
+        type = YtelseType.ORDINÆR_KONTANTSTØTTE,
+        sats = satsperiode.sats,
+        prosent = satsperiode.prosent,
+    )
+}
+
+private fun validerBeregnetPeriode(
+    beløpsperiode: SatsPeriode,
+    behandlingId: Long,
+) {
+    if (beløpsperiode.fom.isAfter(beløpsperiode.tom)) {
```

Comment:
Valideringen her blir ikke truffet av enhetstester, om det ikke er vanskelig kan det være greit å skrive en test for validringen 

<img width="628" alt="image" src="https://github.com/user-attachments/assets/e9b56d15-5226-430e-b6f8-b2ac50c7496b">